clarity/src/vm/analysis/type_checker/v2_1/tests/contracts.rs

```diff
@@ -3540,3 +3543,121 @@ fn clarity_trait_experiments_cross_epochs(
         res => panic!("expected success, got {res:?}"),
     };
 }
+
+/// Pass various types to `contract-hash?`
+#[apply(test_clarity_versions)]
+fn test_contract_hash(#[case] version: ClarityVersion, #[case] epoch: StacksEpochId) {
+    let sources = [
+        "(contract-hash? 'ST1PQHQKV0RJXZFY1DGX8MNSNYVE3VGZJSRTPGZGM.foo)",
+        "(contract-hash? 'ST1PQHQKV0RJXZFY1DGX8MNSNYVE3VGZJSRTPGZGM)",
+        "(contract-hash? 123)",
+        "(contract-hash? u123)",
+        "(contract-hash? true)",
+        "(contract-hash? 0x1234)",
+        "(contract-hash? \"60 percent of the time, it works every time\")",
+        "(contract-hash? u\"I am serious, and don't call me Shirley.\")",
+        "(contract-hash? (list 1 2 3))",
+        "(contract-hash? { a: 1, b: u2 })",
+        "(contract-hash? (some u789))",
+        "(contract-hash? (ok true))",
+    ];
+    let results = if version >= ClarityVersion::Clarity4 {
+        [
+            Ok(Some(
+                TypeSignature::new_response(
+                    TypeSignature::SequenceType(SequenceSubtype::BufferType(
+                        BufferLength::try_from(32u32).unwrap(),
+                    )),
+                    TypeSignature::UIntType,
+                )
+                .unwrap(),
+            )),
+            Ok(Some(
```

Comment:
Why does `(contract-hash? 'ST1PQHQKV0RJXZFY1DGX8MNSNYVE3VGZJSRTPGZGM)` pass type-checking at all?  Can't the type-checker pass determine that the given principal is not a contract?